### ISSUE TITLE
Discussion: issue template 

### ISSUE THREAD

**Issue by tabrindle** [styled-components/styled-components#1584]
In reported issue after reported issue, I see users having trouble with their environment rather than styled-components. Even I have done this: turns out I had multiple versions installed, or was using an old version. Using an old version of safari, or maybe installing with a buggy version of npm. This isn't your fault, but takes up your time to resolve. 

I'd like to update your issue template to make use of my [envinfo](https://www.npmjs.com/package/envinfo) package instead of a manual copypasta from a package.json.

Here is an example of what you can do:

`$ npx envinfo@next --system --binaries --browsers --npmPackages styled-components,babel-plugin-styled-components --markdown --duplicates --clipboard`

This creates a diagnostic report like this:

## System:
 - OS: macOS High Sierra 10.13
 - CPU: x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
 - Free Memory: 633.63 MB
 - Total Memory: 16.00 GB
 - Shell: /usr/local/bin/bash - 4.4.12
## Binaries:
 - Node: 8.9.4
 - Yarn: 1.3.2
 - npm: 5.6.0
 - Watchman: 4.9.0
## Browsers:
 - Chrome: 65.0.3325.146
 - Chrome Canary: 67.0.3364.0
 - Firefox: 58.0
 - Firefox Developer Edition: 57.0
 - Firefox Nightly: 58.0a1
 - Safari: 11.0
 - Safari Technology Preview: 11.1
### styled-components:
 - wanted: ^3.1.6
 - installed: 3.1.6

This is of course all configurable, and can do as little or as much as you want, formatted in markdown, yaml or json. Other projects have already made use of this in their command line tools - create-react-app, react-native and expo all use envinfo under the hood to much success. 

Let me know what you think!

**Comment by probablyup** [comment]
Love it! Please send a PR :)

**Comment by tabrindle** [comment]
Yeah that's an awesome idea - you could use that cli to white label envinfo for this report as well. Envinfo has a programmatic API as well that is a little more powerful/specific. I only didn't propose that because you don't already have a cli that Im aware of. 

Depending on your bandwidth, I can send a PR for the `npx envinfo` change for now while you work on the PoC?

Also since we are getting more in the realm of dedicated tools - I'm compelled to mention a related project to envinfo - [solidarity](https://github.com/infinitered/solidarity). Its a project based environment checker that is very powerful, and also uses envinfo under the hood. That could also be a route here.

**Comment by mxstbr** [comment]
I'd love for you kick-start the CLI effort too @tabrindle—a quick fix to add envinfo to the templates would be much appreciated though :pray:

**Comment by kitten** [comment]
@tabrindle love to have a PR for that :+1:

**Comment by tabrindle** [comment]
Sweet. @kitten get the repo started and add me, I can add envinfo stub functionality for reference.

**Comment by tabrindle** [comment]
I also updated the readme in envinfo to include some example usage: https://github.com/tabrindle/envinfo#programmatic-usage

We should be able to do pretty much exactly this:

```
import envinfo from 'envinfo';

console.log(
  envinfo.run(
    {
      System: ['OS', 'CPU'],
      Binaries: ['Node', 'Yarn', 'npm'],
      Browsers: ['Chrome', 'Firefox', 'Safari'],
      npmPackages: ['styled-components', 'babel-plugin-styled-components'],
    }
  )
);
```
